src/match/recursiveMatch.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ArrayType, Criterion } from '..';
+
+import nativeMatch from './nativeMatch';
+
+/**
+ * RecursiveMatch.
+ *
+ * @param element - ArrayType.
+ * @param criterium - Criterion.
+ * @param keys - String[].
+ * @param options - Object.
+ * @param options.ignorePaths - RegExp[].
+ * @param options.pathAlias - { [s: string]: RegExp }.
+ * @returns Boolean.
+ */
+export default function recursiveMatch(
+  element: ArrayType | string | number | any,
```

Comment:
I think there shouldn't be an `any` here

Reply:
Yes i'm going to use typescript recursion

Reply:
fixed

Reply:
fixed can you check please the new type